azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/ImplementationInstance.kt

```diff
@@ -188,14 +188,15 @@ class JarImplementationInstance<
 
             override fun run() {
                 val implementations = Implementation.locateImplementations<AF, CF, QF, AGF>()
+                    .filter { it.name.split("-").first() == implementationName }
                 result = if (implementations.size == 1) {
                     val implementation = implementations.single()
                     Log.debug("Running task type={} implementation={}", taskType, implementation.name)
                     val implementationTask = TaskWrapper(taskType, implementation, task)
                     implementationTask.run(scenario)
                 } else {
                     val exception =
-                        IllegalStateException("Implementation instance jar should contain exactly one implementation, but ${implementations.size} found")
+                        IllegalStateException("Implementation instance jar should contain exactly one implementation with base name ${implementationName}, but ${implementations.size} found")
```

Comment:
:no_entry_sign:  I don't understand what 'base' means and I think it can be omitted